### PR TITLE
Experiment Aggregator merge and destroy states in batch

### PR DESCRIPTION
**Pull request by kitaisreal**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
This is an automated comment for commit 1217b5e8284a61100ccbdd7e2562d30fa04d4cea with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/53589/1217b5e8284a61100ccbdd7e2562d30fa04d4cea/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Interpreters/Aggregator.cpp [hunk 1]
```diff
@@ -2479,48 +2479,21 @@ void NO_INLINE Aggregator::mergeDataNullKey(
...
         mergeDataNullKey<Method, Table>(table_dst, table_src, arena);
 
+    PaddedPODArray<AggregateDataPtr> dst_places;
+    PaddedPODArray<AggregateDataPtr> src_places;
+
     auto merge = [&](AggregateDataPtr & __restrict dst, AggregateDataPtr & __restrict src, bool inserted)
     {
         if (!inserted)
         {
-#if USE_EMBEDDED_COMPILER
```
Comment (nickitat) [inline src/Interpreters/Aggregator.cpp:2493]: maybe it also worth to disable jit in configs, so the old binary also won't use it

**Comment by kitaisreal** [comment]
Performance improvements:
https://s3.amazonaws.com/clickhouse-test-reports/53589/a2b9cefd06fa281599b71b72a63595e7655c73c4/performance_comparison_aarch64_[1_4]/report.html
https://s3.amazonaws.com/clickhouse-test-reports/53589/a2b9cefd06fa281599b71b72a63595e7655c73c4/performance_comparison_aarch64_[2_4]/report.html
https://s3.amazonaws.com/clickhouse-test-reports/53589/a2b9cefd06fa281599b71b72a63595e7655c73c4/performance_comparison_aarch64_[3_4]/report.html

**Review by KochetovNicolai — APPROVED** [review]
I did not check git, but changes in Aggregator looks fine.

**Pull request merged**